percy/percy-web
# Remove snapshot id param (aka: goodbye auto-scrolling behavior)

Removing the snapshot id query parameter solves both scroll-related issues in clubhouse.
https://app.clubhouse.io/percy/story/1594/auto-scrolling-is-broken-in-builds-with-a-mix-of-approved-unapproved-snapshots
https://app.clubhouse.io/percy/story/1599/if-click-to-approve-a-snapshot-then-scroll-down-to-the-next-one-without-waiting-for-the-approval-once-the-first-one-is-approved

Removing this means that clicking in a snapshot will not generate a unique URL, and navigating to a URL with a snapshot id query param will not scroll to that snapshot anymore. But we can still get a unique URL from a snapshot by either using the nice copy-link button or copying the full-screen page url.

The keyboard nav for hopping up and down snapshots still works, including for builds that have more than 150 diffs (where they are all collapsed). Hopping up and down via keyboard still scrolls the page to the correct snapshot.

However, to make the keyboard nav work, there still needs to be some scroll behavior. I've rewritten the scroll behavior to be less code and make more sense (IMO).

This *does not* fix the bug currently in production where, when closing a full screen snapshot, it does _not_ scroll the the correct position for that snapshot.

## QA suggestions:
Click on some snapshots. Does what you expect happen? Click approve, click some more snapshots, scroll and click some snapshots, use the arrow keys to navigate, open the no-diff batch of snapshots and click/keyboard nav around there.
